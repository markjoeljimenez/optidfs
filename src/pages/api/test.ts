import { XMLParser } from 'fast-xml-parser';
import { NextApiRequest, NextApiResponse } from 'next';
import { getToken } from 'next-auth/jwt';

export default async function GET(req: NextApiRequest, res: NextApiResponse) {
	const token = await getToken({ req });

	const response = await fetch(
		`https://fantasysports.yahooapis.com/fantasy/v2/users;use_login=1/games`,
		{
			headers: { Authorization: `Bearer ${token?.accessToken}` },
		}
	);

	res.status(200).json(new XMLParser().parse(await response.text()));
}
