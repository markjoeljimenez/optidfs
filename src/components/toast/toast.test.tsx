import { render } from '@/test/render';
import userEvent from '@testing-library/user-event';
import Toast from './toast';

describe('Toast', () => {
	it('should display toast error', () => {
		// Arrange
		const toast = render(
			<Toast visible id="1">
				Test toast
			</Toast>
		);

		// Assert
		expect(toast).toMatchSnapshot();
	});

	it('should remove toast error on close button click', () => {
		// Arrange
		const toast = render(
			<Toast visible id="1">
				Test toast
			</Toast>
		);

		// Act
		const toastCloseButton = toast.getByRole('button');
		userEvent.click(toastCloseButton);

		// Assert
		// Need to set timeout because it takes .25 of a second for the toast to be removed from the DOM
		setTimeout(() => {
			expect(toast.container.firstChild).not.toBeInTheDocument();
		}, 500);
	});
});
