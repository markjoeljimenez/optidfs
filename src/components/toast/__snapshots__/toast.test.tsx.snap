// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should display toast error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex p-4 mb-4 bg-red-100 rounded-lg dark:bg-red-200 animate-enter"
        id="1"
        role="alert"
      >
        <svg
          class="flex-shrink-0 w-5 h-5 text-red-700 dark:text-red-800"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="mx-3 text-sm font-medium text-red-700 dark:text-red-800"
        >
          Test toast
        </div>
        <button
          aria-label="Close"
          class="ml-auto -mx-1.5 -my-1.5 bg-red-100 text-red-500 rounded-lg focus:ring-2 focus:ring-red-400 p-1.5 hover:bg-red-200 inline-flex h-8 w-8 dark:bg-red-200 dark:text-red-600 dark:hover:bg-red-300"
          data-dismiss-target="1"
          role="button"
          type="button"
        >
          <span
            class="sr-only"
          >
            Close
          </span>
          <svg
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex p-4 mb-4 bg-red-100 rounded-lg dark:bg-red-200 animate-enter"
      id="1"
      role="alert"
    >
      <svg
        class="flex-shrink-0 w-5 h-5 text-red-700 dark:text-red-800"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="mx-3 text-sm font-medium text-red-700 dark:text-red-800"
      >
        Test toast
      </div>
      <button
        aria-label="Close"
        class="ml-auto -mx-1.5 -my-1.5 bg-red-100 text-red-500 rounded-lg focus:ring-2 focus:ring-red-400 p-1.5 hover:bg-red-200 inline-flex h-8 w-8 dark:bg-red-200 dark:text-red-600 dark:hover:bg-red-300"
        data-dismiss-target="1"
        role="button"
        type="button"
      >
        <span
          class="sr-only"
        >
          Close
        </span>
        <svg
          class="w-5 h-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
